src/ParallelDownloader.php

```diff
@@ -244,7 +244,7 @@ protected function getRemoteContents($originUrl, $fileUrl, $context, array &$res
             return $result;
         }
 
-        if (!$this->downloader) {
+        if (!preg_match('/^https?:/', $originUrl) || !$this->downloader) {
```

Comment:
I would swap the conditions. Checking for a `null` downloader is cheaper than running the regex.

Reply:
True, changed.